### ISSUE TITLE
what is our preferred recommendation for installing Nightscout

### ISSUE THREAD

**Issue by bewest** [nightscout/nightscout.github.io#8]
This is a FAQ/POLL that needs to be ongoing.

The current website instructions allow us to track who has done them by looking at [forks](/../../cgm-remote-monitor/network/members) of [cgm-remote-monitor](/../../cgm-remote-monitor)

Braindump: I would like to augment instructions for other providers, including for @jasoncalabrese 's project glu, especially if we can automate instructions.

Braindump, with support of community, it would be great to wrap the whole thing up in a service which does the whole thing.
##

**Comment by bewest** [comment]
## What does free mean with Nightscout?

Overall, I'm not sure it's wise to set up expectation that it's free to operate or run Nightscout.  While it is true that Nightscout is open source and always will be, from the core developer's perspective we can't control pricing of interoperable medical devices, servers, database and storage, we can only provide the source code.  I've chatted with some of the other core developers, and I'm not sure everyone is comfortable with the Nightscout-is-a-free-product-for-free idea. The reasoning behind distribution on a zero-cost basis was originally to mitigate risk to the authors as an act of "free speech" and conveniently resonated with #PayItForward, but never a raison d'etre or requirement.  In fact the news coverage when Nightscout first came out cited the cost at around $150 to include usb cables, a compatible android phone, etc.

Nightscout has grown in complexity and maturity that require resources to sustain, especially if we are talking about uninterrupted working Nightscout for years on end.  Most of the trial accounts provided have quotas on the resources used that will impact the Nightscout experience in some negative way without payment.  Most developers pay for their servers to stay up and for enough storage to never run out and recommend for their friends to do the same for the best experience.  Sometimes we see groups of folks rent a common server and split it up among themselves. Unfortunately, none of those sales go back to benefit Nightscout, and the complexity itself is a limiting factor to who can benefit from Nightscout.  I really appreciate calling out how Nightscout is open source, is an act of free speech and liberates time away from diabetes and back on life.  I do wonder if the total cost of ownership might perhaps deserve it's own section discussing the tactics and pros and cons separate from instructions of how to get things done?  As developers, we've done what we can to provide different ways to get things done, including ways to be mindful of cost, but our ability to impact third party hosting providers and constantly changing quotas and pricing models is quite limited.  What are some of the values we want to express in selecting our preferred recommendation for how to install Nightscout?

* Operates at low cost?
* Operates reliably?  Will operate for a long time?
* Simple to set up?
* Fewer steps to accomplish?
* Security and safety for end user vs the community as a whole?
* Easy to debug and troubleshoot - over chat/messenger?

**Comment by psonnera** [comment]
Whilst this all makes sense, I believe most users are still attracted by the free aspect of Nightscout. Times have changed a lot from the original project environment that was filling a huge gap on data sharing with the G4. From my limited experience we have mainly four types of new users (but I'm probably missing others).
1. Those who are on FGM+transmitter and need data sharing not provided by the transmitter vendor app.
2. Those who want a smartwatch/gadget displaying their BG real time.
3. Those who don't have devices compatible with the vendors app and need to use open source.
4. Close loop users for remote operation and advanced features of Nightscout.
We have followed with fear and interest the evolution of the third party hosting from Azure to Heroku and now from mLab to Atlas and trust developers to be able to mitigate the fast changing cloud ecosystem resources with minimum impact on a free Nightscout.
I agree that some form of warning should be added to the documentation. This will be done.
Your statement on #PayItForward makes me very uncomfortable, so all this https://www.nightscoutfoundation.org/ was a scam?
Whilst some already proposed paid private hosting servers (only one comes to my mind now -but there are surely more-  jakicukier.pl) others took the free private hosting servers, like 10be.de, probably unsustainable in the long run.
Since you own and promote t1pal, it is obvious that your recommendations go in favor of a paid hosted service.
Now, if you want to look at it as it is now, the third party hosting service operates at low cost unless you need large storage, it is rather reliable (not been there for a long time but Heroku is very rarely down) but I agree as the original disclaimer says, that you shouldn't rely on it, it is complex to setup -no doubt- but many Facebook groups and individuals outside Facebook provide very efficient support.
Bottom line: I'll update the documentation and trust you, developers to leave the possibility of a DIY Nightscout solution.

**Comment by sulkaharo** [comment]
I would add to above any users who want to see insulin and CGM data in glanceable fashion, which includes all loopers but also people uploading their 600-series pumps. NS is not just for the advanced use of looping systems, but actually the only solution in market to remotely see both insulin dosing and CGM data in real time remotely.

I fully agree with the notion that users taking "Nightscout is free" for granted is probably hurting the ecosystem. A decent portion of the issues people need support on are related to using free services for hosting, instead of spending even a few dollars per month on hosting services. As an example, I'm personally hosting my database in a $5/mo Digital Ocean droplet, which has happily chugged along for years without running out of capacity, vs people on free mLab running out of space all the time, which has required both a ton of support but also feature development in NS so it's easier for non-technical people to delete data. The big value for paying _some_ money for NS can be to skip the parts that go wrong with installation and no hassle updates & capacity. Incidentally this also means less need to support users, where the support community has also seen a lot of people basically burn out from the endless amount of work supporting new users. Same goes for developers, where developing / maintaining NS is anything but trivial, but frequently users feel entitled to make demands and take offence if someone implies their wishes as customers are not met.

Re Nightscout Foundation - I don't have the actual data but AFAIK >95% of the resources from the Foundation have been used to market Nightscout. The organisation is not participating in the development of NS, other than occasionally giving financial support. I know the foundation has paid for some services (like GitHub) and in some cases paid for testing devices, but AFAIK this spend is a small percentage of the Foundation's budget. 

So - I personally would point people toward T1Pal, but also actually recommend that instead of the free hosting services they would spend the small amount of money per month to get the smallest paid tiers where appropriate. Like on Heroku, actually spending the $7 per month to get the Hobbyist dyno that doesn't sleep, instead of installing uptime services to ping the dyno to prevent sleeping.

**Comment by psonnera** [comment]
Thanks @sulkaharo for these important and technical considerations. I will update the documentation starting page accordingly.

**Comment by Cheun1** [comment]
Hello je suis votre conversation qui est très constructive ,et je tiens à vous dire bravo.
